packages/create/bin/create-stacks-workspace.spec.ts

```diff
@@ -0,0 +1,219 @@
+import chalk from 'chalk';
+import enquirer from 'enquirer';
+import yargs from 'yargs';
+
+import {
+    determineAppName,
+    // eslint-disable-next-line unicorn/prevent-abbreviations
+    determineE2eTestRunner,
+    determinePreset,
+    determineRepoName,
+    getConfiguration,
+} from './create-stacks-workspace';
+import { CreateStacksArguments, Preset } from './types';
+
+const mockExit = jest
+    .spyOn(process, 'exit')
+    .mockImplementation(() => undefined as never);
+
+const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation();
+
+describe('determine preset', () => {
+    it('valid preset', async () => {
+        const result = await determinePreset({
+            preset: 'next',
+        } as yargs.Arguments<CreateStacksArguments>);
+
+        expect(result).toBe('next');
+    });
+
+    it('invalid preset', async () => {
+        jest.spyOn(enquirer, 'prompt').mockResolvedValueOnce({
+            Preset: 'apps',
+        });
+
+        await determinePreset({
+            preset: 'invalid',
+        } as yargs.Arguments<CreateStacksArguments>);
+
+        expect(consoleErrorSpy).toBeCalledWith(
+            chalk.red`Invalid preset: It must be one of the following: apps,ts,next`,
+        );
+
+        expect(mockExit).toHaveBeenCalledWith(1);
+    });
+
+    it('no preset', async () => {
+        const result = await determinePreset({
+            name: 'testpreset',
+        } as yargs.Arguments<CreateStacksArguments>);
+
+        expect(result).toBe('apps');
+    });
+
+    it('interactive set with no preset', async () => {
+        const result = await determinePreset({
+            name: 'testpreset',
+            interactive: false,
+        } as yargs.Arguments<CreateStacksArguments>);
+
+        expect(result).toBe('apps');
+    });
+});
+
+describe('determine repo name', () => {
+    it('valid repo name no position arg', async () => {
+        const result = await determineRepoName({
+            _: [],
+            name: 'myrepo',
+        } as unknown as yargs.Arguments<CreateStacksArguments>);
```

Comment:
Just wondering about the unknown type cast here, is this needed?

Reply:
Currently required as we're passing in a positional argument just for this test. Probably need to look at how this is handled in the future, to avoid casting if possible.